Add comments to strings to help translators

devinreams opened the pull request:
Fixes #204

https://github.com/mozilla-lockbox/lockbox-android/blob/204-initial-string-comments/app/src/main/res/values/strings.xml

## To Do

- [x] double check the original issue to confirm it is fully satisfied
- [x] add testing notes and screenshots in PR description to help guide reviewers
- [x] make sure CI builds are passing (e.g.: fix lint and other errors)

Pike commented on app/src/main/res/values/strings.xml:
```diff
@@ -5,24 +5,35 @@
+    <!-- This string is the full and formal title of the application, used in areas like a header or welcome screen -->
     <string name="app_name">Firefox Lockbox</string>
+    <!-- This is the short name or label of the application, used as the name in the Android app launcher -->
     <string name="app_label">Lockbox</string>
```
I moved these two into a separate file in #250, as they're not supposed to be localized, per our conversation with Sandy.

Pike commented on app/src/main/res/values/strings.xml:
```diff
@@ -5,24 +5,35 @@
+    <!-- This string is used as the label on a button on a first-run welcome screen to sign in and start using the application -->
     <string name="welcome_start_btn">Get Started</string>
+    <!-- This string provides context near the "welcome_start_btn" to describe what using the application will require -->
     <string name="welcome_instructions">To use Firefox Lockbox, you\'ll need a\nFirefox Account with saved logins.</string>
```
Can we pass in the brandname here? If not, the localization note should be explicit about not allowing to modify `Firefox Lockbox`. Similar for the other strings containing the lockbox brand.

Also, should we use proper curly apostrophe in our copy on Lockbox? We do in all/most Firefox products.

Delphine replied:
+1 to using curly apostrophe.

Concerning the brand name: can't we just use something like %1$s which we use for other mobile Android product names?

Pike commented on app/src/main/res/values/strings.xml:
```diff
@@ -5,24 +5,35 @@
-    <string name="lock_now">Lock Now</string>
     <string name="all_entries_a_z">All entries (A-Z)</string>
     <string name="all_entries_recent">All entries (Recent)</string>
     <string name="search_your_entries" tools:ignore="ExtraTranslation">search your entries</string>
```
What does tools:ignore do?

oyintoboss commented:
Ok

devinreams commented:
>So for now I put in quick disclaimers to not translate "Firefox Lockbox" or "Firefox Account".

After chatting with the team, this is probably the best/simplest solution for now as opposed to trying to pass multiple string parameters in everywhere...

@mozilla-lockbox/mobile-engineering @Pike @Delphine would you mind a re-review of this PR, please?

devinreams commented on app/src/main/res/values/strings.xml:
```diff
@@ -5,96 +5,183 @@
     <string name="unlock_button">Unlock</string>
-    <string name="lock_now">Lock Now</string>
+
+    <!-- This message appears in a toast informing the user it is currently downloading and syncing entry data -->
```
>It may be worth defining very simply/briefly what "entries" here refers to exactly, since localizers will not all familiar with this project

@Delphine OK! I added a quick message to the first occurrence. Is it necessary to repeat it throughout?

devinreams commented on app/src/main/res/values/strings.xml:
```diff
@@ -5,96 +5,183 @@
+    <!-- This is the title of a dialog requesting permission for the application to unlock using the device fingerprint sensor -->
+    <string name="enable_fingerprint_dialog_title">Enable Fingerprint</string>
+    <!-- This is the message requesting permission for the application to unlock using the device fingerprint sensor -->
+    <string name="enable_fingerprint_dialog_subtitle">Give Lockbox permission to unlock this app with the fingerprint sensor.</string>
```
>better to be consistent and use "Firefox Lockbox" everywhere

Great catch! Thank you...

devinreams commented:
Thank you for the review @Delphine! I addressed your comments and just one remains, I believe:

>@Pike: For plurals in iOS, I know we have to format them differently because there is no proper plural support (so the workaround would be to do minutes: 1, minutes: 2, etc).

Have you run into this on Android yet, @Pike?

I'm hoping to merge this in soon so I don't get too far behind `master` and so that new strings get their own nice descriptions written during development. If we want to tackle numeric translations separately we can just spin out a separate issue to discuss further...

Thanks again @Delphine @Pike!

Pike commented on app/src/main/res/values/strings.xml:
```diff
@@ -5,96 +5,183 @@
+    <!-- This is the label a screenreader uses to inform the user they selected the auto lock setting toggle -->
+    <string name="auto_lock_description">Auto lock setting</string>
+    <!-- This is a time duration for the automatic application locking setting equal to 1 minute -->
+    <string name="one_minute">1 minute</string>
```
Android itself has plural support, but compare-locales and pontoon don't support that yet.

In this case, enumerating the values is also good enough.

Sorry for the lag in getting back to this, I agree with merging and follow-ups.
